src/api/routes/nlp.router.ts

```
@@ -62,16 +63,16 @@ router.post("/process", async (req: Request, res: Response) => {
       try {
         const db = getWalmartDatabaseManager();
-        for (const item of result.items.slice(0, 5)) { // Limit to 5 items
-          const searchResults = await db.walmartProducts.searchProducts(item, 3);
+        for (const item of result?.items?.slice(0, 5)) { // Limit to 5 items
+          const searchResults = await db?.walmartProducts?.searchProducts(item, 3);
           products.push(...searchResults);
         }
         
         // Notify WebSocket about product matches
-        if (sessionId && products.length > 0) {
+        if (sessionId && products?.length || 0 > 0) {
```

Biome (2.1.2)

[error] Comparing to itself is potentially pointless.



(lint/suspicious/noSelfCompare)